src/ObjCRuntime/Runtime.cs

```diff
@@ -1735,11 +1737,31 @@ internal static IntPtr GetProtocolForType (Type type)
 			throw new ArgumentException ($"'{type.FullName}' is an unknown protocol");
 		}
 
-		[BindingImpl (BindingImplOptions.Optimizable)]
 		internal static bool IsUserType (IntPtr self)
 		{
 			var cls = Class.object_getClass (self);
+			lock (usertype_cache) {
+#if NET
+				var result = CollectionsMarshal.GetValueRefOrAddDefault (usertype_cache, cls, out var exists);
```

Comment:
I think this is needed to fix the test failures:

```suggestion
				ref var result = CollectionsMarshal.GetValueRefOrAddDefault (usertype_cache, cls, out var exists);
```